File: src/main/java/com/how2java/service/DiaryService.java
package com.how2java.service;

import java.util.List;

import com.how2java.pojo.Diary;
import com.how2java.pojo.entity.PageResult;

public interface DiaryService {

	  public Long add(Diary  diary) throws Exception;  
	       
	      
	    public void delete(int id);  
	       
	      
	    public Diary  get(int id) throws Exception;  
	     
	      
	    public int update(Diary  category) throws Exception;   
	       
	      
	    public List<Diary > list(String name);
	    
	      
	    public int count();


		PageResult findByPage(int pageNum, int pageSize,String name);


		public void delete(int[] ids);  

		public String lastEdit();


		public int countMy(String name);
}

File: src/main/webapp/js/service/diaryService.js
// 定义服务层:
app.service("diaryService",function($http){
	this.findAll = function(){
		return $http.get("../diary/findAll.do");
	}
	
	this.findByPage = function(page,rows){
		return $http.get("../diary/findByPage.do?page="+page+"&rows="+rows);
	}
	
	this.save = function(entity){
		return $http.post("../diary/save.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../diary/update.do",entity);
	}
	
	this.findById=function(id){
		return $http.get("../diary/findById.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../diary/delete.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../diary/search.do?page="+page+"&rows="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../diary/selectOptionList.do");
	}
	
	this.count = function(){
		return $http.get("../diary/count.do");
	}
	
	this.lastEdit = function(){
		return $http.get("../diary/lastEdit.do");
	}
	
	this.countMy = function(){
		return $http.get("../diary/countMy.do");
	}
	
	this.afterSave = function(entity){
		return $http.post("../diary/afterSave.do",entity);
	}

});